packages/server-wallet/src/protocols/open-channel.ts

```diff
@@ -17,7 +17,11 @@ import {
   CompleteObjective,
 } from './actions';
 
-export type ProtocolState = {app: ChannelState; ledgerFundingRequested?: boolean};
+export type ProtocolState = {
+  app: ChannelState;
+  ledgerFundingRequested?: boolean;
+  ledger?: ChannelState;
+};
```

Comment:
This seems better as a union type: Something like
```
export type AppState = {
  app: ChannelState;
};
export type DirectlyFundedAppState = AppState & {
  type: 'DirectFundingProtocolState';
};
export type LedgerFundedAppState = AppState & {
  type: 'LedgerFundingProtocolState';
  ledgerFundingRequested: boolean;
  ledger: ChannelState;
};
export type ProtocolState = DirectlyFundedAppState | LedgerFundedAppState;
```

Reply:
0cf44d1cb